### PR TITLE
Extract reward cell to its own nib

### PR DESCRIPTION
**Pull request by mbrandonw**
Been tinkering with how we can reuse table view cells across multiple controllers/storyboards, and turns out it's very easy. So, I took a shot at doing this for our `RewardCell`, which is definitely our most complicated cell. I did this one cause some future work is gonna bring rewards into some new views.

To do this I literally just cut+pasted the cell from the storyboard to a brand new `*.xib` file. Then I have to do an explicit `self.tableView.register(nib)` step in the controller because that's what the storyboard + prototype cell does for you. But after that it just worked!

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/ProjectPamphletContentViewController.swift [hunk 1]
```diff
@@ -28,6 +28,9 @@ public final class ProjectPamphletContentViewController: UITableViewController {
       self, action: #selector(scrollViewPanGestureRecognizerDidChange(_:))
     )
 
+    self.tableView.register(UINib(nibName: "RewardCell", bundle: Bundle.framework),
+                            forCellReuseIdentifier: "RewardCell")
```
Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/ProjectPamphletContentViewController.swift]: might be nice to put this in a new enum in `Storyboard.swift`?

File: Kickstarter-iOS/Library/Nib.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import UIKit
+
+public enum Nib: String {
+  case RewardCell
+}
+
+extension UITableView {
+  public func register(nib: Nib, inBundle bundle: Bundle = .framework) {
+    self.register(UINib(nibName: nib.rawValue, bundle: bundle), forCellReuseIdentifier: nib.rawValue)
+  }
```
Comment (theginbin) [inline Kickstarter-iOS/Library/Nib.swift]: nice nice!

Comment (mbrandonw) [inline Kickstarter-iOS/Library/Nib.swift]: great suggestion!

Comment (theginbin) [inline Kickstarter-iOS/Library/Nib.swift]: you know how much i love enums

**Pull request merged**